libs/lwt_platform/unix/Lwt_platform.ml

```diff
@@ -26,5 +26,9 @@
 let run = Lwt_main.run
 let detach = Lwt_preemptive.detach
 let init_preemptive = Lwt_preemptive.init
-let set_engine () = Lwt_engine.set (new Lwt_engine.libev ())
+
+let set_engine () =
+  if Sys.unix then Lwt_engine.set (new Lwt_engine.libev ())
```

Comment:
you can factorize Lwt_engine.set outside the if (I was having trouble seeing the difference between the 2 branches)

Reply:
Agreed that it's a little gross, but I hit a compilation error when I tried it this way: (might be a noob issue)
![CleanShot 2024-01-09 at 08 12 45@2x](https://github.com/semgrep/semgrep/assets/47152/1ec4a107-d55a-460e-aa8a-30a1daf9af7b)

Reply:
oh it's because of the use of objects. The type of Lwt_engine.set is quite complex,
`val set : ?transfer : bool -> ?destroy : bool -> #t -> unit`
this #t is the type of a class, so you'll probably need some casting operations to reconcile both branches.
Simpler to do what you did indeed.